tfx/dependencies.py

```diff
@@ -262,17 +262,12 @@ def make_extra_packages_docs() -> list[str]:
     list[str]
         List of packages required for building docs
     """
-    return [
-        "mkdocs",
-        "mkdocstrings[python]",
-        "mkdocs-material",
-        "griffe-inherited-docstrings",
-        "mkdocs-autorefs",
-        "mkdocs-jupyter",
-        "mkdocs-caption",
-        "pymdown-extensions",
-        "markdown-grid-tables",
-    ]
+    with open(Path(__file__).resolve().parent.parent / "requirements-docs.txt", "r") as fp:
+        reqs = fp.readlines()
+
+    reqs = [req.replace("\n", "") for req in reqs]
+
+    return reqs
```

Comment:
```suggestion
    return [req.replace("\n", "") for req in reqs]
```

Can just return this, I think. Also, doesn't `\n` need to be replaced with `' '`?

In the future, we are now using `pyproject.toml` so if we can it would be useful to convert all dependencies to dynamic metadata to make `setuptools` responsible for loading and handling this (and other dependency groups): https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata

Apparently this will not work for just one optional dependency - all must be specified dynamically - but anyway it's a much cleaner mechanism.

Reply:
> Can just return this, I think. Also, doesn't `\n` need to be replaced with `' '`?

I'm not sure it makes sense to have a space `' '` appended to the end of every package name:

```python
In [1]: with open("requirements-docs.txt", "r") as fp:
   ...:     reqs = fp.readlines()
   ...: 

In [2]: reqs
Out[2]: 
['mkdocs\n',
 'mkdocstrings[python]\n',
 'mkdocs-material\n',
 'griffe-inherited-docstrings\n',
 'mkdocs-autorefs\n',
 'mkdocs-jupyter\n',
 'mkdocs-caption\n',
 'markdown-grid-tables\n']
```

I was trying to dynamically reproduce what was there before:

```python
def make_extra_packages_docs() -> list[str]:
    """Get a list of packages required for building docs as HTML.

    Returns
    -------
    list[str]
        List of packages required for building docs
    """
    return [
        "mkdocs",
        "mkdocstrings[python]",
        "mkdocs-material",
        "griffe-inherited-docstrings",
        "mkdocs-autorefs",
        "mkdocs-jupyter",
        "mkdocs-caption",
        "pymdown-extensions",
        "markdown-grid-tables",
    ]
```

Reply:
Whoops, I see that you're using `readlines()` - makes sense!